narwhals/_spark_like/dataframe.py

```diff
@@ -112,17 +115,53 @@ def _from_native_frame(self: Self, df: DataFrame) -> Self:
     def columns(self: Self) -> list[str]:
         return self._native_frame.columns  # type: ignore[no-any-return]
 
-    def collect(self: Self) -> PandasLikeDataFrame:
-        import pandas as pd  # ignore-banned-import()
+    def collect(
+        self: Self,
+        backend: ModuleType | Implementation | str | None,
+        **kwargs: Any,
+    ) -> CompliantDataFrame:
+        if backend in (None, "pandas", Implementation.PANDAS, get_pandas()):
+            import pandas as pd  # ignore-banned-import
+
+            from narwhals._pandas_like.dataframe import PandasLikeDataFrame
+
+            return PandasLikeDataFrame(
+                native_dataframe=self._native_frame.toPandas(),
+                implementation=Implementation.PANDAS,
+                backend_version=parse_version(pd.__version__),
+                version=self._version,
+            )
 
-        from narwhals._pandas_like.dataframe import PandasLikeDataFrame
+        elif backend in ("pyarrow", Implementation.PYARROW, get_pyarrow()):
+            import pyarrow as pa  # ignore-banned-import
 
-        return PandasLikeDataFrame(
-            native_dataframe=self._native_frame.toPandas(),
-            implementation=Implementation.PANDAS,
-            backend_version=parse_version(pd.__version__),
-            version=self._version,
-        )
+            from narwhals._arrow.dataframe import ArrowDataFrame
+
+            return ArrowDataFrame(
+                native_dataframe=pa.Table.from_batches(
+                    self._native_frame._collect_as_arrow()
+                ),
+                backend_version=parse_version(pa.__version__),
+                version=self._version,
+            )
+
+        elif backend in ("polars", Implementation.POLARS, get_polars()):
+            import polars as pl  # ignore-banned-import
+            import pyarrow as pa  # ignore-banned-import
+
+            from narwhals._polars.dataframe import PolarsDataFrame
+
+            return PolarsDataFrame(
+                df=pl.from_arrow(  # type: ignore[arg-type]
+                    pa.Table.from_batches(self._native_frame._collect_as_arrow())
```

Comment:
Ritchie answer in [stackoverflow](https://stackoverflow.com/a/73205690/12411536)